kore/src/Kore/Step.hs

```diff
@@ -163,13 +169,40 @@ anyRewrite
 anyRewrite rewrites =
     Strategy.any (rewriteStep <$> rewrites)
 
+priorityAllStrategy
+    :: ToRulePattern rewrite
+    => [rewrite]
+    -> Strategy (Prim rewrite)
+priorityAllStrategy rewrites =
+    Strategy.any (fmap allRewrites priorityGroups)
```

Comment:
Is `Strategy.any` the right choice here? @traiansf

Reply:
Also, if we replaced `allRewrites` with `anyRewrite`, then wouldn't this be a valid implementation of `priorityAnyStrategy`? If that is the case, could we parameterize this function by the "inner" strategy, so that we have a single implementation? Or in other words, make this a combinator?

Reply:
> Is Strategy.any the right choice here?

I think that it serves the purpose, at least if we assume the initial configuration to be concrete (i.e., no possibility of remainders).  My argument for this choice is that, at least in its current implementation, any would try each group in sequence, until it finds one matching; if there are multiple matches in that group, it will explore all of them, but would not try rules in subsequent groups.

If we would like to think about symbolic search instead, then we would probably end up with something similar to the all-path strategy.

> Or in other words, make this a combinator?

We used to have it written as a combinator, but we weren't sure there would be other uses for it; note also that sorting seems more natural than grouping for `priorityAnyStrategy`.  

But, of course, if you think it might be useful to have it as a combinator, we'll make it so :-)

Reply:
If we are going to keep Strategy.any here, should we rename it to Strategy.first, or something like that? We could also add an any function that just calls first.